fix(request-validator): fix content-type breaking change (#5421)
Kong/docs.konghq.com: app/_hub/kong-inc/request-validator/_index.md

      description: |
        The request body schema specification. One of `body_schema` or `parameter_schema`
        must be specified.
    - name: allowed_content_types # latest version of param
      minimum_version: "3.1.x"
      required: true
      default:
        - application/json
      value_in_examples: null
      datatype: Set of string elements
      description: |
        List of allowed content types. The value can be configured with the `charset` parameter. For example, `application/json; charset=UTF-8`.

        {:.note}
        > **Notes:** 
        * Body validation is only done for `application/json` and skipped for any other allowed content types. 
        * Only one parameter is supported. If a request sends more than one parameter with the Content-Type header, only the first parameter is evaluated and the rest are truncated.

>         * From `3.1.x`, The parameter is checked forcibly, which means a request with parameter(e.g. `application/json; charset=UTF-8`) is **NOT** to be considered valid for the one without the parameter(e.g. `application/json`).

Reviewer: This bit is only visible "from 3.1.x" so probably not necessary to include the version in this note.

Reviewer 2: Agreed. I'll adjust it.

>         * From `3.1.x`, The parameter is checked forcibly, which means a request with parameter(e.g. `application/json; charset=UTF-8`) is **NOT** to be considered valid for the one without the parameter(e.g. `application/json`).

Reviewer 2: ```suggestion
        * The parameter is checked for an exact match, which means a request with a parameter (e.g. `application/json; charset=UTF-8`) is **NOT** considered valid for one without the same parameter (e.g. `application/json`).
```
How's this? "The parameter is checked forcibly" is slightly awkward phrasing.

Author: Using “exact match” may not be accurate since the charset parameter was treated as case-insensitive. 
ChatGPT told me another phase is  "the parameter is strictly validated"

Reviewer 2: I like it. Thanks, ChatGPT 😁

    - name: allowed_content_types
      maximum_version: "2.7.x"
      required: true
      default:
        - application/json
      value_in_examples: null
      datatype: Set of string elements
      description: |
        List of allowed content types. 
        
        {:.note}
        > **Note:** Body validation is only
        done for `application/json` and skipped for any other allowed content types.
    - name: allowed_content_types
      minimum_version: "2.8.x"
      maximum_version: "2.8.x"
      required: true
      default:
        - application/json
      value_in_examples: null
      datatype: Set of string elements
      description: |
          List of allowed content types. The value can be configured with the `charset` parameter. For example, `application/json; charset=UTF-8`.
          
          {:.note}
          > **Notes:** 
          * Body validation is only done for `application/json` and skipped for any other allowed content types. 
          * Only one parameter is supported. If a request sends more than one parameter with the Content-Type header, only the first parameter is evaluated and the rest are truncated.

>           * application/json` doesn't match with `application/json; charset=UTF-8`.

Author: ```suggestion

```

        The request body schema specification. One of `body_schema` or `parameter_schema`
        must be specified.
    - name: allowed_content_types

>       minimum_version: "3.1.x"

Reviewer 2: I'm confused. Which release are all these changes going into, exactly?

Author: There are a couple of things

- content-type with parameter was introduced at v2.8 and > v3.1
- fixing breaking behavior at v2.8
- 3.0 is a gap(for some reason) that as same as version < v2.7  (I found the v3.0.x is showing the 2.8.x's description, it seems like the (minimum_version: "3.0.x"  and maximum_version: "3.0.x" does not work)

Reviewer 2: Very weird. Luckily, we'll have this type of thing under control soon, as we're switching to generating these docs from plugin schemas in the near future :) Thanks for the fixes!

Author: LGTM. I made another small change.
